libraries/holochain-service-client/src/main/java/org/holochain/androidserviceruntime/holochain_service_client/HolochainServiceClient.kt

```diff
@@ -49,42 +50,126 @@ class HolochainServiceClient(
     }
 
     /// Install an app
-    fun installApp(payload: InstallAppPayloadFfiParcel): AppInfoFfiParcel {
-        // Call installApp on service
-        return this.mService!!.installApp(payload)
+    suspend fun installApp(payload: InstallAppPayloadFfiParcel): AppInfoFfiParcel {
+        Log.d(TAG, "installApp")
+
+        val deferred = CompletableDeferred<AppInfoFfiParcel>()
+        var callbackBinder = object : IHolochainServiceCallbackStub() {
```

Comment:
Create a new IPC binder for the callback service and pass it in as an input parameter to the HolochainService call.